agent/stats/container.go

```diff
@@ -96,6 +97,9 @@ func (container *StatsContainer) collect() {
 func (container *StatsContainer) processStatsStream() error {
 	dockerID := container.containerMetadata.DockerID
 	seelog.Debugf("Collecting stats for container %s", dockerID)
+	if container.client == nil {
+		return errors.New("Client is not set.")
```

Comment:
It'd be good add some more context for the error message here: `"container stats stream: client is not set"`. Also, when can this error occur? Was there a code path where this was not being set?